src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -388,7 +438,23 @@ void ISocketOutput.Write(ArraySegment<byte> buffer, bool immediate, bool chunk)
 
         Task ISocketOutput.WriteAsync(ArraySegment<byte> buffer, bool immediate, bool chunk, CancellationToken cancellationToken)
         {
-            return WriteAsync(buffer, immediate, chunk);
+            if (cancellationToken.IsCancellationRequested)
+            {
+#if DOTNET5_4
+                return Task.FromCanceled(cancellationToken);
+#else
+                throw new TaskCanceledException();
```

Comment:
This is a significant behavioral difference between the two frameworks. Isn't it possible to do something similar to what's being done for dotnet5.4?